_posts/en/2024-10-09-wallet-integration-guide.md

```diff
@@ -0,0 +1,590 @@
+unnecessary force closes during fee spikes for years now. With removal of this
+protocol message, commitment transactions could be set to a static 1 sat/vbyte
+feerate. With TRUC transactions, we can ensure that competing commitment
+transactions with anchor spends are allowed to RBF each other over the network,
+and if there are competing output spends from the same commitment transaction,
+that RBF can occur no matter which output is being spent. TRUC transactions are
+also allowed to be 0-fee, allowing reduction in spec complexity. With TRUC’s
+sibling eviction, we can also drop the 1 block CSV locktimes, since we are no
+longer overly concerned with what unconfirmed outputs are being spent, as long
+as each party can spend a single output themselves.
+
+With TRUC + P2A Anchors, we can reduce the blockspace usage of the current two
+anchors down to a single keyless anchor. This anchor requires no commitment to a
+public key or  signatures, saving additional blockspace. The fee bumping can
+also be outsourced to other agents that have no privileged key material. Anchors
+could also consist of a single output with shared key material between the
+counterparties rather than P2A, at the cost of additional vbytes in the benign
+unilateral close case.
+
+Similar strategies can be pursued when implementing advanced features such as
+splicing, to reduce the risk of RBF pinning. For example, a TRUC channel splice
+that is less than 1kvB in size could CPFP another channel's unilateral close,
+without exposing the bumper to RBF pins. Subsequent bumps can be done in series
+by replacing just the channel splice transaction. This comes at the cost of
+revealing the TRUC transaction type during splices.
+
+As you can see, significant complexity can be avoided and savings achieved with
+the updated features, provided each transaction can fit into the 1P1C paradigm.
+
+### Ark
+
+Not all transaction patterns fit into the 1P1C paradigm. A prime example of this
+is [Ark][topic ark] outputs, which commit to a tree of presigned(or covenant committed)
+transactions to unroll a shared UTXO.
+
+Ideally the initial unilateral close of an Ark tree would be:
+
+1. The publication of an entire merkle branch to the underlying virtual
+UTXO(vUTXO)
+2. Each of these transactions are 0-fee, to avoid fee prediction or the
+requirement to decide who pays fees a priori
+3. The ultimate leaf transaction has a 0-value anchor spend where the CPFP
+pays for the entire merkle tree’s publication to a miner’s mempool and inclusion
+in a block
+
+To execute this ideal properly, we are missing a few things:
+
+1. General package relay. We have no method currently of propagating these
+chains of fee-less transactions over the P2P network robustly.
+2. If too many branches are published at low feerates, this can result in
+users being unable to publish their own branch promptly due to cluster limit
+pinning. This could be disastrous as the number of counterparties becomes large,
```

Comment:
You haven't defined cluster limit, and cluster limits don't exist yet. Maybe describe descendant limit pinning, and add a sidenote that the same issue persists even with cluster mempool

Reply:
@instagibbs